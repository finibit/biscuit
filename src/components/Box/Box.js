import React from 'react'
import styled from 'styled-components'
import styles from '../../styles'
import types from '../../types'

const BoxStyled = styled.div`
  box-sizing: border-box;

  ${styles.compose(
    styles.color,
    styles.background,
    styles.border,
    styles.shadow,
    styles.space,
    styles.layout,
    styles.position,
    styles.flexbox
  )};
`

const Box = (props) => (
  <BoxStyled
    {...props}
  />
)

Box.propTypes = {
  ...types.base,
  ...types.color,
  ...types.background,
  ...types.border,
  ...types.shadow,
  ...types.space,
  ...types.layout,
  ...types.position,
  ...types.flexbox
}

Box.defaultProps = {
  children: null,
  as: 'div'
}

export default Box
