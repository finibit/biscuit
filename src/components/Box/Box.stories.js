import React from 'react'
import { storiesOf } from '@storybook/react'

import { Theme, Box } from '../../../src'

storiesOf('Box', module)
	.add('default', () => (
		<Theme>
			<Box>
				This is a default Box component
			</Box>
		</Theme>
	))
	.add('as', () => (
		<Theme>
			<Box as="section">
				Uses `section` instead of `div` as DOM element
			</Box>
		</Theme>
	))
	.add('color', () => (
		<Theme>
			<Box color="info">
				Background color set to value from theme.
			</Box>
			<Box color="#cafeee">
				Background color set to custom value.
			</Box>
		</Theme>
	))
	.add('margin', () => (
		<Theme>
			<Box margin="md" color="light-0">
				The amount of margin is the same for all sides.
			</Box>
			<Box margin={{ vertical: 'xl', horizontal: 'md' }} color="light-0">
				The amount of margin is different vertically and horizontally.
			</Box>
			<Box margin={['xs', 'sm', 'md', { vertical: 'xl', horizontal: 'md' }]} color="light-0">
				The amount of margin changes at each responsive breakpoint.
			</Box>
		</Theme>
	))
	.add('padding', () => (
		<Theme>
			<Box padding="md" color="light-0">
				The amount of padding is the same for all sides.
			</Box>
			<Box padding={{ vertical: 'xl', horizontal: 'md' }} color="light-1">
				The amount of padding is different vertically and horizontally.
			</Box>
			<Box padding={['xs', 'sm', 'md', { vertical: 'xl', horizontal: 'md' }]} color="light-2">
				The amount of padding changes at each responsive breakpoint.
			</Box>
		</Theme>
	))
	.add('width', () => (
		<Theme>
			<Box width="300px" color="light-0">
				300px wide
			</Box>
		</Theme>
	))
	.add('height', () => (
		<Theme>
			<Box height="300px" color="light-0">
				300px tall
			</Box>
		</Theme>
	))
	.add('overflow', () => (
		<Theme>
			<Box overflow="auto" width="100px" height="100px">
				The behavior of auto overflow depends on the browser.
			</Box>
			<Box overflow="scroll" width="100px" height="100px">
				Scroll bars will be added to the box to allow for scrolling the contents that is too long to fit into the box.
			</Box>
			<Box overflow="hidden" width="100px" height="100px">
				No scroll bars will be added and the content will be clipped to the size of the box.
			</Box>
			<Box overflow="visible" width="100px" height="100px">
				No scroll bars will be added and the content will overflow the box.
			</Box>
		</Theme>
	))
	.add('direction', () => (
		<Theme>
			<Box direction="row" padding="md" color="light-0">
				<Box padding="md" color="light-1">
					The first box is on the left hand side of the second one.
				</Box>
				<Box padding="md" color="light-1">
					The second box is on the right hand side of the first one.
				</Box>
			</Box>
			<Box direction="row-reverse" padding="md" color="light-0">
				<Box color="light-1" padding="md">
					First box in reversed row direction
				</Box>
				<Box color="light-1" padding="md">
					Second box in reversed row direction
				</Box>
			</Box>
			<Box direction="col" padding="md" color="light-0">
				<Box color="light-1" padding="md">
					First box in column direction
				</Box>
				<Box color="light-2" padding="md">
					Second box in column direction
				</Box>
			</Box>
			<Box direction="col-reverse" padding="md" color="light-0">
				<Box color="light-1" padding="md">
					First box in reversed column direction
				</Box>
				<Box color="light-2" padding="md">
					Second box in reversed column direction
				</Box>
			</Box>
		</Theme>
	))
	.add('wrap', () => (
		<Theme>
			<Box wrap>
				<Box padding="md" color="light-0">
					This is the first box in a wrapping container
				</Box>
				<Box padding="md" color="light-1">
					This is the second box in a wrapping container
				</Box>
				<Box padding="md" color="light-2">
					This is the third box in a wrapping container
				</Box>
			</Box>
		</Theme>
	))
	.add('order', () => (
		<Theme>
			<Box>
				<Box order={3} padding="md" color="light-0">
					The first box but is displayed as the last one
				</Box>
				<Box order={2} padding="md" color="light-1">
					The second box is displayed in the middle
				</Box>
				<Box order={1} padding="md" color="light-2">
					The last box but is displayed as the first one
				</Box>
			</Box>
		</Theme>
	))
	.add('justify', () => (
		<Theme>
			<Box justify="center" width="400px" color="light-0">
				<Box padding="md" color="light-1">
					Box #1
				</Box>
				<Box padding="md" color="light-1">
					Box #2
				</Box>
			</Box>
		</Theme>
	))
	.add('align', () => (
		<Theme>
			<Box height="500px" align="center" color="light-0">
				<Box padding="md" color="light-1">
					First box
				</Box>
				<Box padding="md" color="light-2">
					Second box
				</Box>
			</Box>
		</Theme>
	))
	.add('alignSelf', () => (
		<Theme>
			<Box direction="row">
				<Box height="300px" color="light-0" padding="md">
					First box with auto alignment
				</Box>
				<Box height="100px" color="light-1" padding="md" alignSelf="center">
					Second box with center alignment
				</Box>
				<Box height="300px" color="light-2" padding="md">
					Third box with auto alignment
				</Box>
			</Box>
		</Theme>
	))
	.add('alignContent', () => (
		<Theme>
			<Box direction="row" width="400px" height="400px" wrap alignContent="evenly" color="light-0">
				<Box color="light-1" padding="md" width="200px">
					First box
				</Box>
				<Box color="light-2" padding="md" width="200px">
					Second box
				</Box>
				<Box color="light-3" padding="md" width="200px">
					Third box
				</Box>
			</Box>
		</Theme>
	))
	.add('css', () => (
		<Theme>
			<Box width="100px" height="100px" css="border: 1px solid #000;" />
		</Theme>
	))
