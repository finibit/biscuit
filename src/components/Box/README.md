### Usage

#### Space

```jsx
<Box color={6} margin={2} padding={2} />
```

#### Size

```jsx
<Box color={6} width="50%" height="50px" />
```

#### Color

```jsx
<Box padding={1} color={6} />
```
```jsx
<Box padding={1} color="#dcba98" />
```
```jsx
<Box padding={1} color="bisque" />
```

### Border

```jsx
<Box padding={1} border={0} />
```
```jsx
<Box padding={1} border={{ width: '1px', style: 'dashed', color: 0 }} />
```

### Elevation

```jsx
<Box padding={1} color={6} elevation={1} />
```
```jsx
<Box padding={1} color={6}  elevation={2} />
```
```jsx
<Box padding={1} color={6}  elevation={3} />
```
```jsx
<Box padding={1} color={6}  elevation={4} />
```
```jsx
<Box padding={1} color={6}  elevation={6} />
```

#### Overflow

```jsx
<Box color={6} height="40px" overflow="auto">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Box>
```
```jsx
<Box color={6} height="40px" overflow="scroll">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Box>
```
```jsx
<Box color={6} height="40px" overflow="hidden">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Box>
```
```jsx
<Box color={6} height="40px" overflow="visible">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Box>
```

#### Direction

```jsx
<Box direction="row">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="column">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```

#### Wrap

```jsx
<Box direction="row" width="10%" wrap>
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```

#### Order

```jsx
<Box direction="row" width="10%">
  <Box margin={1} padding={1} color={6} order={3}>1</Box>
  <Box margin={1} padding={1} color={6} order={2}>2</Box>
  <Box margin={1} padding={1} color={6} order={1}>3</Box>
</Box>
```
#### Justify

```jsx
<Box direction="row" justify="start">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" justify="end">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" justify="center">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" justify="between">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" justify="around">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" justify="evenly">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={1} color={6}>2</Box>
  <Box margin={1} padding={1} color={6}>3</Box>
</Box>
```

#### Align

```jsx
<Box direction="row" align="start">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={3} color={6}>2</Box>
  <Box margin={1} padding={2} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" align="end">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={3} color={6}>2</Box>
  <Box margin={1} padding={2} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" align="center">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={3} color={6}>2</Box>
  <Box margin={1} padding={2} color={6}>3</Box>
</Box>
```
```jsx
<Box direction="row" align="stretch">
  <Box margin={1} padding={1} color={6}>1</Box>
  <Box margin={1} padding={3} color={6}>2</Box>
  <Box margin={1} padding={2} color={6}>3</Box>
</Box>
```

#### Css
```jsx
import { css } from 'styled-components';

const style = css`
  transform: skewX(-20deg);
`;

<Box color={6} height="40px" css={style} />
```
