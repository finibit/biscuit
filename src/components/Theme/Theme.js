import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
