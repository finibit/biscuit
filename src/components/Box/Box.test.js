import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Box from './Box'

describe('Box', () => {
	test('default', () => {
		const component = renderer.create((
			<Theme>
				<Box />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with margin', () => {
		const component = renderer.create((
			<Theme>
				<Box m={1} />
				<Box m={32} />
				<Box m="5%" />
				<Box mt={1} />
				<Box mt={32} />
				<Box mt="5%" />
				<Box ml={1} />
				<Box ml={32} />
				<Box ml="5%" />
				<Box mb={1} />
				<Box mb={32} />
				<Box mb="5%" />
				<Box mr={1} />
				<Box mr={32} />
				<Box mr="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with padding', () => {
		const component = renderer.create((
			<Theme>
				<Box p={1} />
				<Box p={32} />
				<Box p="5%" />
				<Box pt={1} />
				<Box pt={32} />
				<Box pt="5%" />
				<Box pl={1} />
				<Box pl={32} />
				<Box pl="5%" />
				<Box pb={1} />
				<Box pb={32} />
				<Box pb="5%" />
				<Box pr={1} />
				<Box pr={32} />
				<Box pr="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with size', () => {
		const component = renderer.create((
			<Theme>
				<Box size="sm" />
				<Box size={64} />
				<Box size="5%" />
				<Box width="sm" />
				<Box width={64} />
				<Box width="5%" />
				<Box height="sm" />
				<Box height={64} />
				<Box height="5%" />
				<Box minWidth={1} />
				<Box minWidth={64} />
				<Box minWidth="5%" />
				<Box minHeight={1} />
				<Box minHeight={64} />
				<Box minHeight="5%" />
				<Box maxWidth={1} />
				<Box maxWidth={32} />
				<Box maxWidth="5%" />
				<Box maxHeight={1} />
				<Box maxHeight={32} />
				<Box maxHeight="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with overflow', () => {
		const component = renderer.create((
			<Theme>
				<Box overflow="hidden" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with color background', () => {
		const component = renderer.create((
			<Theme>
				<Box bg="primary" />
				<Box bg="#012345" />
				<Box bg="cyan" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with image background', () => {
		const component = renderer.create((
			<Theme>
				<Box backgroundImage="url('https://picsum.photos/128')" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with border', () => {
		const component = renderer.create((
			<Theme>
				<Box border="primary" />
				<Box border="1px solid #012345" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with shadow', () => {
		const component = renderer.create((
			<Theme>
				<Box boxShadow="md" />
				<Box boxShadow="10px 10px 5px 0px rgba(0,0,0,0.75)" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with position', () => {
		const component = renderer.create((
			<Theme>
				<Box position="relative" />
				<Box top={1} />
				<Box top={32} />
				<Box top="5%" />
				<Box left={1} />
				<Box left={32} />
				<Box left="5%" />
				<Box bottom={1} />
				<Box bottom={32} />
				<Box bottom="5%" />
				<Box right={1} />
				<Box right={32} />
				<Box right="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with z-index', () => {
		const component = renderer.create((
			<Theme>
				<Box zIndex={1} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('is flex', () => {
		const component = renderer.create((
			<Theme>
				<Box flex="1 1 auto" />
				<Box alignItems="center" />
				<Box alignContent="center" />
				<Box justifyContent="center" />
				<Box flexWrap="wrap" />
				<Box flexBasis="auto" />
				<Box flexDirection="column" />
				<Box justifySelf="center" />
				<Box alignSelf="center" />
				<Box order="2" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})
})
