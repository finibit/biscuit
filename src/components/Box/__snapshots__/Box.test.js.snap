// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box default 1`] = `
.c0 {
  box-sizing: border-box;
}

<div>
  <div
    className="c0"
  />
</div>
`;

exports[`Box is flex 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  box-sizing: border-box;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c3 {
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  box-sizing: border-box;
  justify-self: center;
}

.c8 {
  box-sizing: border-box;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c9 {
  box-sizing: border-box;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
    order="2"
  />
</div>
`;

exports[`Box with border 1`] = `
.c0 {
  box-sizing: border-box;
  border: 1px solid #ffdd59;
}

.c1 {
  box-sizing: border-box;
  border: 1px solid #012345;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
</div>
`;

exports[`Box with color background 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #ffdd59;
}

.c1 {
  box-sizing: border-box;
  background-color: #012345;
}

.c2 {
  box-sizing: border-box;
  background-color: cyan;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;

exports[`Box with image background 1`] = `
.c0 {
  box-sizing: border-box;
  background-image: url('https://picsum.photos/128');
}

<div>
  <div
    className="c0"
  />
</div>
`;

exports[`Box with margin 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0.375rem;
}

.c1 {
  box-sizing: border-box;
  margin: 32px;
}

.c2 {
  box-sizing: border-box;
  margin: 5%;
}

.c3 {
  box-sizing: border-box;
  margin-top: 0.375rem;
}

.c4 {
  box-sizing: border-box;
  margin-top: 32px;
}

.c5 {
  box-sizing: border-box;
  margin-top: 5%;
}

.c6 {
  box-sizing: border-box;
  margin-left: 0.375rem;
}

.c7 {
  box-sizing: border-box;
  margin-left: 32px;
}

.c8 {
  box-sizing: border-box;
  margin-left: 5%;
}

.c9 {
  box-sizing: border-box;
  margin-bottom: 0.375rem;
}

.c10 {
  box-sizing: border-box;
  margin-bottom: 32px;
}

.c11 {
  box-sizing: border-box;
  margin-bottom: 5%;
}

.c12 {
  box-sizing: border-box;
  margin-right: 0.375rem;
}

.c13 {
  box-sizing: border-box;
  margin-right: 32px;
}

.c14 {
  box-sizing: border-box;
  margin-right: 5%;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
  <div
    className="c13"
  />
  <div
    className="c14"
  />
</div>
`;

exports[`Box with overflow 1`] = `
.c0 {
  box-sizing: border-box;
  overflow: hidden;
}

<div>
  <div
    className="c0"
    overflow="hidden"
  />
</div>
`;

exports[`Box with padding 1`] = `
.c0 {
  box-sizing: border-box;
  padding: 0.375rem;
}

.c1 {
  box-sizing: border-box;
  padding: 32px;
}

.c2 {
  box-sizing: border-box;
  padding: 5%;
}

.c3 {
  box-sizing: border-box;
  padding-top: 0.375rem;
}

.c4 {
  box-sizing: border-box;
  padding-top: 32px;
}

.c5 {
  box-sizing: border-box;
  padding-top: 5%;
}

.c6 {
  box-sizing: border-box;
  padding-left: 0.375rem;
}

.c7 {
  box-sizing: border-box;
  padding-left: 32px;
}

.c8 {
  box-sizing: border-box;
  padding-left: 5%;
}

.c9 {
  box-sizing: border-box;
  padding-bottom: 0.375rem;
}

.c10 {
  box-sizing: border-box;
  padding-bottom: 32px;
}

.c11 {
  box-sizing: border-box;
  padding-bottom: 5%;
}

.c12 {
  box-sizing: border-box;
  padding-right: 0.375rem;
}

.c13 {
  box-sizing: border-box;
  padding-right: 32px;
}

.c14 {
  box-sizing: border-box;
  padding-right: 5%;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
  <div
    className="c13"
  />
  <div
    className="c14"
  />
</div>
`;

exports[`Box with position 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
}

.c1 {
  box-sizing: border-box;
  top: 0.375rem;
}

.c2 {
  box-sizing: border-box;
  top: 32px;
}

.c3 {
  box-sizing: border-box;
  top: 5%;
}

.c4 {
  box-sizing: border-box;
  left: 0.375rem;
}

.c5 {
  box-sizing: border-box;
  left: 32px;
}

.c6 {
  box-sizing: border-box;
  left: 5%;
}

.c7 {
  box-sizing: border-box;
  bottom: 0.375rem;
}

.c8 {
  box-sizing: border-box;
  bottom: 32px;
}

.c9 {
  box-sizing: border-box;
  bottom: 5%;
}

.c10 {
  box-sizing: border-box;
  right: 0.375rem;
}

.c11 {
  box-sizing: border-box;
  right: 32px;
}

.c12 {
  box-sizing: border-box;
  right: 5%;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
</div>
`;

exports[`Box with shadow 1`] = `
.c0 {
  box-sizing: border-box;
  box-shadow: 0 3px 6px rgba(0,0,0,0.08),0 3px 6px rgba(0,0,0,0.11);
}

.c1 {
  box-sizing: border-box;
  box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  />
</div>
`;

exports[`Box with size 1`] = `
.c0 {
  box-sizing: border-box;
  width: 32px;
  height: 32px;
}

.c1 {
  box-sizing: border-box;
  width: 64px;
  height: 64px;
}

.c2 {
  box-sizing: border-box;
  width: 5%;
  height: 5%;
}

.c3 {
  box-sizing: border-box;
  width: 32px;
}

.c4 {
  box-sizing: border-box;
  width: 64px;
}

.c5 {
  box-sizing: border-box;
  width: 5%;
}

.c6 {
  box-sizing: border-box;
  height: 32px;
}

.c7 {
  box-sizing: border-box;
  height: 64px;
}

.c8 {
  box-sizing: border-box;
  height: 5%;
}

.c9 {
  box-sizing: border-box;
  min-width: 1px;
}

.c10 {
  box-sizing: border-box;
  min-width: 64px;
}

.c11 {
  box-sizing: border-box;
  min-width: 5%;
}

.c12 {
  box-sizing: border-box;
  min-height: 1px;
}

.c13 {
  box-sizing: border-box;
  min-height: 64px;
}

.c14 {
  box-sizing: border-box;
  min-height: 5%;
}

.c15 {
  box-sizing: border-box;
  max-width: 1px;
}

.c16 {
  box-sizing: border-box;
  max-width: 32px;
}

.c17 {
  box-sizing: border-box;
  max-width: 5%;
}

.c18 {
  box-sizing: border-box;
  max-height: 1px;
}

.c19 {
  box-sizing: border-box;
  max-height: 32px;
}

.c20 {
  box-sizing: border-box;
  max-height: 5%;
}

<div>
  <div
    className="c0"
    size="sm"
  />
  <div
    className="c1"
    size={64}
  />
  <div
    className="c2"
    size="5%"
  />
  <div
    className="c3"
    width="sm"
  />
  <div
    className="c4"
    width={64}
  />
  <div
    className="c5"
    width="5%"
  />
  <div
    className="c6"
    height="sm"
  />
  <div
    className="c7"
    height={64}
  />
  <div
    className="c8"
    height="5%"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
  <div
    className="c13"
  />
  <div
    className="c14"
  />
  <div
    className="c15"
  />
  <div
    className="c16"
  />
  <div
    className="c17"
  />
  <div
    className="c18"
  />
  <div
    className="c19"
  />
  <div
    className="c20"
  />
</div>
`;

exports[`Box with z-index 1`] = `
.c0 {
  box-sizing: border-box;
  z-index: 1;
}

<div>
  <div
    className="c0"
  />
</div>
`;
